### PR TITLE
add LuaJIT port

### PR DESCRIPTION
**Pull request by cat4anna**

### CODE REVIEW

**Comment by pgrabas** [comment]
I forgot to add, LuaJIT port conflicts with existing Lua port, so one should not have installed both.

File: ports/luajit/001-fixStaticBuild.patch [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ @if "%1"=="amalg" goto :AMALGDLL
+ @if "%1"=="static" goto :STATIC
+@@ -76,6 +77,7 @@ buildvm -m folddef -o lj_folddef.h lj_opt_fold.c
+ @if errorlevel 1 goto :BAD
+ @goto :MTDLL
+ :STATIC
++@if "%DEBUG%"=="1" set LJCOMPILE=%LJCOMPILE% /MTd
+ %LJCOMPILE% lj_*.c lib_*.c
+ @if errorlevel 1 goto :BAD
+ %LJLIB% /OUT:%LJLIBNAME% lj_*.obj lib_*.obj
```
Comment (codicodi) [inline ports/luajit/001-fixStaticBuild.patch]: Could you extend this patch to make debug dll also link to debug runtime (/`MDd`)?

**Comment by codicodi** [comment]
The change looks good, but the patch does not apply correctly now (git and vcpkg say that it did, but `src/msvcbuild.bat` is only partially modified) and static build fails. Could you regenerate it once more?

**Comment by pgrabas** [comment]
I have regenerated the patch. Applying now works as should.

**Comment by ras0219-msft** [comment]
LGTM, thanks for the PR and thanks @codicodi for the review!

**Comment by kklouzal** [comment]
Is there a way to tell vcpkg to compile LuaJIT as C++? @pgrabas @ras0219-msft @codicodi 

Thank you for your time.
P.S.
I cannot thank you guys enough for this entire project!

**Comment by pgrabas** [comment]
I don't think there is any way to do it without modifying build scripts. Do you have any issues with interoperability with c++?

**Pull request merged**